src/dpdkstat.c

```diff
@@ -0,0 +1,671 @@
+#define DPDKSTAT_MAX_BUFFER_SIZE (4096*4)
+#define DPDK_SHM_NAME "dpdk_collectd_stats_shm"
+#define REINIT_SHM 1
+#define RESET 1
+#define NO_RESET 0
+
+enum DPDK_HELPER_ACTION {
+  DPDK_HELPER_ACTION_COUNT_STATS,
+  DPDK_HELPER_ACTION_SEND_STATS,
+};
+
+enum DPDK_HELPER_STATUS {
+  DPDK_HELPER_NOT_INITIALIZED = 0,
+  DPDK_HELPER_WAITING_ON_PRIMARY,
+  DPDK_HELPER_INITIALIZING_EAL,
+  DPDK_HELPER_ALIVE_SENDING_STATS,
+  DPDK_HELPER_GRACEFUL_QUIT,
+};
+
+struct dpdk_config_s {
+  /* General DPDK params */
+  char coremask[DATA_MAX_NAME_LEN];
+  char memory_channels[DATA_MAX_NAME_LEN];
+  char socket_memory[DATA_MAX_NAME_LEN];
+  char process_type[DATA_MAX_NAME_LEN];
+  char file_prefix[DATA_MAX_NAME_LEN];
+  cdtime_t interval;
+  uint32_t eal_initialized;
+  uint32_t enabled_port_mask;
+  uint32_t eal_argc;
+  /* Helper info */
+  int   collectd_reinit_shm;
+  pid_t helper_pid;
+  sem_t sema_helper_get_stats;
+  sem_t sema_stats_in_shm;
+  int   helper_pipes[2];
+  enum DPDK_HELPER_STATUS helper_status;
+  enum DPDK_HELPER_ACTION helper_action;
+  /* xstats info */
+  uint32_t num_ports;
+  uint32_t num_xstats;
+  cdtime_t port_read_time[RTE_MAX_ETHPORTS];
+  uint32_t num_stats_in_port[RTE_MAX_ETHPORTS];
+  struct rte_eth_link link_status[RTE_MAX_ETHPORTS];
+  struct rte_eth_xstats xstats;
+  /* rte_eth_xstats from here on until the end of the SHM */
+};
+typedef struct dpdk_config_s dpdk_config_t;
+
+static int g_configured = 0;
+static dpdk_config_t *g_configuration = 0;
+
+static int dpdk_config_init_default(void);
+static int dpdk_config(oconfig_item_t *ci);
+static int dpdk_helper_init_eal(void);
+static int dpdk_helper_run(void);
+static int dpdk_helper_spawn(enum DPDK_HELPER_ACTION action);
+static int dpdk_init (void);
+static int dpdk_read(user_data_t *ud);
+static int dpdk_shm_cleanup(void);
+static int dpdk_shm_init(size_t size);
+void module_register(void);
+
+/* Write the default configuration to the g_configuration instances */
+static int dpdk_config_init_default(void)
+{
+    g_configuration->interval = plugin_get_interval();
+    WARNING("dpdkstat: No time interval was configured, default value %lu ms is set\n",
+             CDTIME_T_TO_MS(g_configuration->interval));
+    g_configuration->enabled_port_mask = 0;
+    g_configuration->eal_argc = 2;
+    g_configuration->eal_initialized = 0;
+    ssnprintf(g_configuration->coremask, DATA_MAX_NAME_LEN, "%s", "0xf");
+    ssnprintf(g_configuration->memory_channels, DATA_MAX_NAME_LEN, "%s", "1");
+    ssnprintf(g_configuration->process_type, DATA_MAX_NAME_LEN, "%s", "secondary");
+    ssnprintf(g_configuration->file_prefix, DATA_MAX_NAME_LEN, "%s",
+             "/var/run/.rte_config");
+  return 0;
+}
+
+static int dpdk_config(oconfig_item_t *ci)
+{
+  int i = 0, ret = 0;
+
+  /* Initialize a POSIX SHared Memory (SHM) object. */
+  dpdk_shm_init(sizeof(dpdk_config_t));
+
+  /* Set defaults for config, overwritten by loop if config item exists */
+  ret = dpdk_config_init_default();
```

Comment:
function changed to
`static void dpdk_config_init_default(void)`
error checking is removed